client/state/happychat/test/selectors.js

```diff
@@ -58,6 +70,59 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( '#canUserSendMessages', () => {
+		const messagingDisabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ABANDONED,
+			HAPPYCHAT_CHAT_STATUS_BLOCKED,
+			HAPPYCHAT_CHAT_STATUS_DEFAULT,
+			HAPPYCHAT_CHAT_STATUS_MISSED,
+			HAPPYCHAT_CHAT_STATUS_PENDING,
+		];
+		const messagingEnabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ASSIGNED,
+			HAPPYCHAT_CHAT_STATUS_ASSIGNING,
+			HAPPYCHAT_CHAT_STATUS_CLOSED,
+			HAPPYCHAT_CHAT_STATUS_NEW,
+		];
+
+		it( 'should return false if Happychat is unavailable', () => {
+			const state = {
+				happychat: {
+					connectionStatus: 'uninitialized',
+					isAvailable: false,
+					chatStatus: HAPPYCHAT_CHAT_STATUS_NEW
+				}
+			};
+			expect( canUserSendMessages( state ) ).to.be.false;
+		} );
+
+		it( "should return false if Happychat is available but the chat status doesn't allow messaging", () => {
+			messagingDisabledChatStatuses.forEach( status => {
+				const state = {
+					happychat: {
+						connectionStatus: 'connected',
+						isAvailable: true,
+						chatStatus: status
+					}
+				};
+				expect( canUserSendMessages( state ) ).to.be.false;
+			} );
+		} );
+
```

Comment:
I'll concede that for right now this is probably worth adding in — but I think longer-term it would be better if these selectors were all in different modules so that if they depend on one another they can be mocked in tests.

That is — `canUserSendMessages` does _not_ care about connection status. It just cares about results from `isHappychatAvailable` and `getHappychatChatStatus`. So in a true unit test on the logic of `canUserSendMessages`, we should be mocking `isHappychatAvailable` and `getHappychatChatStatus` to abstract away their internal logic, and not reaching into implementation details of those dependencies by shaping the state based on their internal logic.

But we can't mock those dependencies when they're all exported from the same module, so for now it's probably fine to add the test you suggested, to cover our bases.